src/bin/cargo-afl.rs

```diff
@@ -91,138 +87,134 @@ fn main() {
 }
 
 #[allow(clippy::too_many_lines)]
-fn clap_app() -> clap::App<'static> {
-    use clap::{
-        App,
-        AppSettings::{
-            AllowExternalSubcommands, AllowHyphenValues, AllowInvalidUtf8ForExternalSubcommands,
-            DisableHelpFlag, DisableHelpSubcommand, DisableVersionFlag, SubcommandRequiredElseHelp,
-        },
-        Arg,
-    };
+fn clap_app() -> clap::Command {
+    use clap::{value_parser, Arg, Command};
 
-    App::new("cargo afl")
-        .bin_name("cargo")
-        .setting(SubcommandRequiredElseHelp)
+    Command::new("cargo afl")
+        .display_name("cargo")
+        .subcommand_required(true)
+        .arg_required_else_help(true)
         .subcommand(
-            App::new("afl")
+            Command::new("afl")
                 .version(crate_version!())
-                .setting(SubcommandRequiredElseHelp)
-                .setting(AllowExternalSubcommands)
-                .setting(AllowInvalidUtf8ForExternalSubcommands)
+                .subcommand_required(true)
+                .arg_required_else_help(true)
+                .allow_external_subcommands(true)
+                .external_subcommand_value_parser(value_parser!(OsString))
                 .override_usage("cargo afl [SUBCOMMAND or Cargo SUBCOMMAND]")
                 .after_help(
                     "In addition to the subcommands above, Cargo subcommands are also \
                  supported (see `cargo help` for a list of all Cargo subcommands).",
                 )
                 .subcommand(
-                    App::new("analyze")
+                    Command::new("analyze")
                         .about("Invoke afl-analyze")
-                        .setting(AllowHyphenValues)
-                        .setting(DisableHelpSubcommand)
-                        .setting(DisableHelpFlag)
-                        .setting(DisableVersionFlag)
+                        .allow_hyphen_values(true)
+                        .disable_help_subcommand(true)
+                        .disable_help_flag(true)
+                        .disable_version_flag(true)
```

Comment:
Needs to be tested, but I believe subcommands now have their version flags disabled by default. So you might no longer need this line.

Reply:
(Ditto for all other subcommands.)

Reply:
I added several tests. Please tell me if you think there are still gaps, or if you do not like their approach(es).

Reply:
Taking a look now.